src/services/cache-service.ts

```diff
@@ -32,9 +32,12 @@ export class CacheService {
         this._cacheContainer[key] = [
             validUntil,
             loadingFunction().then(response => {
-                const cachedResponse = Promise.resolve(response);
-                this._cacheContainer[key] = [validUntil, cachedResponse];
-                return cachedResponse;
+                const cachedResponse = Promise.resolve(response)
+                if (validUntil >= Date.now()) { // lest older, dead key replace newer, still living one
+                    this._cacheContainer[key] = [validUntil, cachedResponse]
+                }
+                // outside 'if', so previous handles on this call still eventually receive the result
```

Comment:
I dont get this comment. The response should always be resolved, even when outdated, not eventually.

Reply:
Yes, "eventually" does mean "we know that it will happen, but not when".

(I added the comment after, originally, putting everything into the `if`, and prompty breaking another test case with it. Double-checking, I had made the same mistake in my original PR #42 , so it seemed like a good case for "Heads-up, tomorrow-Zsar: Better remember this!")